scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -2093,7 +2093,34 @@ class Router(formatOps: FormatOps) {
           if (close.isEmpty) Split.ignored else Split(Space, 1)
 
         case Newlines.classic if style.align.arrowEnumeratorGenerator =>
-          Split(Space, 1).withIndent(StateColumn, expire, After)
+          Split(Space, 1)
```

Comment:
that's probably my fault for not handling this case before, we might want to consider non-classic cases as well.

should we just add this indent once to line 2114 (old) or 2141 (new) to all cases if this flag is on?

Reply:
It works with `fold`.

`unfold` always add newline after `<-`. Looks like it doesn't support this alignment type at all. May be mark them as conflicting?

```
for {
  x <-
    new Integer {
      def value = 2
    }
  y <- Int(2)
  cond <-
    if (variable)
      doSomething
    else
      doAnything
} yield x + y
```

`keep` formats code ugly, it wasn't fixed in https://github.com/scalameta/scalafmt/pull/1795 :(

```
for {
  x <- new Integer {
    def value = 2
  }
  y <- Int(2)
  cond <- if (variable) doSomething
  else doAnything
} yield x + y
```

Reply:
looking at the code, `unfold` doesn't add newline for expressions enclosed in `()` (and all except `keep` do the same for `{}`). don't remember why.

Reply:
unfold with braces:

```
for {
  x <- (
      new Integer {
        def value = 2
      }
  )
  y <- Int(2)
  cond <- (
      if (variable)
        doSomething
      else
        doAnything
  )
} yield x + y
```

I still don't think that the `align.arrowEnumeratorGenerator` is applicable here